src/Metadata/Resource/Factory/UriTemplateResourceMetadataCollectionFactory.php

```diff
@@ -164,6 +168,12 @@ private function configureUriVariables(ApiResource|HttpOperation $operation): Ap
         $variables = $route->getPathVariables();
 
         if (\count($variables) !== \count($uriVariables)) {
+            if ($hasUserConfiguredUriVariables) {
+                $this->logger?->warning('You have configured less URI variables then present in your URI template. You will need to decorate the IRI converter to avoid IRI generation errors.');
```

Comment:
```suggestion
                $this->logger?->warning('You have configured less URI variables than present in your URI template. You will need to decorate the IRI converter to avoid IRI generation errors.');
```

Reply:
I think the last sentence is not clear for most users.

Reply:
I'm wondering what's the best solution here, should I just throw? These use cases are really borderline and should probably just use Symfony. It's definitely wrong to add uriVariables if some are already configured so this needs a fix but whether to throw or to warn idk :|.

Reply:
Maybe just add a larger explanation? For instance: "this is probably an error in your configuration, but if you need to use URI variables for other uses than as identifiers...".

Reply:
then I should throw, even though I like this to be possible. Note that this will throw later in our IriConverter right now. Maybe that actually the error inside the IriConverter is clear enough?